pkg/gds/server.go

```diff
@@ -522,12 +524,17 @@ func (s *Server) Status(ctx context.Context, in *api.HealthCheck) (out *api.Serv
 	// Request another health check between 30-60 min from now
 	now := time.Now()
 
-	// Create the service state as unhealthy unless we have callback streams
+	// Default service state is healthy.
 	out = &api.ServiceState{
 		Status:    api.ServiceState_HEALTHY,
 		NotBefore: now.Add(30 * time.Minute).Format(time.RFC3339),
 		NotAfter:  now.Add(60 * time.Minute).Format(time.RFC3339),
 	}
 
+	// If we're in maintenance mode, update the service state.
+	if s.conf.Maintenance {
+		out.Status = api.ServiceState_MAINTENANCE
+	}
```

Comment:
@rebeccabilbro last minor change since your review - the `status` rpc can report maintenance mode without an error.